examples/tactile_pose_estimation.py

```diff
@@ -108,15 +116,20 @@ def run_learning_loop(cfg):
     # -------------------------------------------------------------------- #
     # Use theseus_layer in an outer learning loop to learn different cost
     # function parameters:
-    measurements = dataset.get_measurements(
-        cfg.train.batch_size, cfg.train.num_batches, time_steps
-    )
-    obj_poses_gt = dataset.obj_poses[0:time_steps, :].clone().requires_grad_(True)
-    eff_poses_gt = dataset.eff_poses[0:time_steps, :].clone().requires_grad_(True)
-    theseus_inputs = {}
-    for _ in range(cfg.train.num_epochs):
+    measurements = dataset.get_measurements(time_steps)
+    for epoch in range(cfg.train.num_epochs):
+        print(" ********************* EPOCH {epoch} *********************")
         losses = []
+        image_idx = 0
         for batch_idx, batch in enumerate(measurements):
```

Comment:
Why is this loop over batch (and later loss.backward inside this loop)? Inner loop would be called only once per outer loop i.e. epoch, no?

Reply:
This is because we can have multiple batches (e.g., 2 batches each with 4 trajectories). If we set `cfg.train.batch_size = cfg.max_episodes`, then inner loop would only be called once. This is analogous to the standard `for batch in dataloader` when training neural nets in `torch`.